public/app/plugins/panel/nodeGraph/useContextMenu.tsx

```diff
@@ -173,24 +179,45 @@ function getItems(links: LinkModel[]) {
   });
 }
 
-function NodeHeader(props: { node: NodeDatum; nodes: DataFrame }) {
-  const index = props.node.dataFrameRowIndex;
-  const fields = getNodeFields(props.nodes);
-  return (
-    <div>
-      {fields.title && <Label field={fields.title} index={index} />}
-      {fields.subTitle && <Label field={fields.subTitle} index={index} />}
-      {fields.details.map((f) => (
-        <Label key={f.name} field={f} index={index} />
-      ))}
-    </div>
-  );
+function NodeHeader({ node, nodes }: { node: NodeDatum; nodes?: DataFrame }) {
+  const index = node.dataFrameRowIndex;
+  if (nodes) {
+    const fields = getNodeFields(nodes);
+
+    return (
+      <div>
+        {fields.title && (
+          <Label
+            label={fields.title.config.displayName || fields.title.name}
+            value={fields.title.values.get(index) || ''}
+          />
+        )}
+        {fields.subTitle && (
+          <Label
+            label={fields.subTitle.config.displayName || fields.subTitle.name}
+            value={fields.subTitle.values.get(index) || ''}
+          />
+        )}
+        {fields.details.map((f) => (
+          <Label key={f.name} label={f.config.displayName || f.name} value={f.values.get(index) || ''} />
+        ))}
+      </div>
+    );
+  } else {
+    // Fallback if we don't have nodes dataFrame. Can happen if we use just the edges frame to construct this.
+    return (
+      <div>
+        {node.title && <Label label={'Title'} value={node.title} />}
```

Comment:
Do we need an empty state/error message if there is not a node title or subtitle?

Reply:
We should always have the node.title here because that will be created from edges source/target IDs which are required.